docs/02-guides/dynamic-routing.mdx

```diff
@@ -0,0 +1,320 @@
+---
+title: Dynamic Routing
+description:
+  This guide explains how you can create dynamic routes for url rewrites in
+  Front-Commerce.
+tags:
+  - routes
+  - dynamic routes
+  - url rewrites
+---
+
+<SinceVersion tag="3.5" />
+
+<p>{frontMatter.description}</p>
+
+## What is dynamic routing?
+
+Dynamic routes are generated based on the data available in your application,
+allowing for more meaningful and SEO-friendly URLs. For example, instead of a
+generic URL such as `/product/sku-6`, you can have a more descriptive URL like
+`/acme-product.html`.
+
+Consider a Remix route for your products defined as `routes/product.$id.tsx`. By
+leveraging dynamic routes, you can replace generic product URLs with
+SEO-friendly alternatives.
+
+#### Generic Route Example
+
+- URL: `/product/sku-6`
+
+#### SEO-Friendly Route Example
+
+- URL: `/acme-product.html`
+
+By implementing dynamic routes, you enhance the readability and search engine
+optimization (SEO) of your URLs, which can improve your site's visibility and
+user experience.
+
+## How to create dynamic routes?
+
+<Steps>
+<Step>
+
+### Create a `DynamicRouteUrlMatcher`
+
+First, define a class that implements the `DynamicRouteUrlMatcher` interface to
+match URLs to your application's routes.
+
+```ts title="extension/UrlMatcher.ts"
+import type {
+  DynamicRouteUrlMatcher,
+  MatchedDynamicRouteUrl,
+} from "@front-commerce/core";
+
+// Define hardcoded routes for demonstration purposes
+const hardcodedRoutes = [
+  {
+    type: "product"
+    path: "/acme-product.html",
+    identifier: "sku-6",
+  },
+];
+
+class UrlMatcher implements DynamicRouteUrlMatcher {
+  matchUrl(path: string): MatchedDynamicRouteUrl | undefined {
+    // Find a matching route in the hardcoded list, ideally this will be from a remote resource
+    const match = hardcodedRoutes.find((route) => route.path === path);
+    if (!match) {
+      return undefined;
+    }
+    return {
+      path: match.path,
+      identifier: match.identifier,
+      params: {
+        id: match.identifier, // The product ID to pass to the route
+      },
+    };
```

Comment:
**issue:** shouldn't it return the `type` too?

In this example there's no relationship between the `handle` type and the matcher.